ecommerce/extensions/payment/processors/payu.py

```diff
@@ -0,0 +1,292 @@
+            'payment_page_url': self.payment_page_url,
+            'merchantId': self.merchantId,
+            'accountId': self.accountId,
+            'ApiKey': self.ApiKey,
+            'referenceCode': basket.order_number,
+            'tax': self.tax,
+            'taxReturnBase': self.taxReturnBase,
+            'currency': basket.currency,
+            'test': self.test,
+            'buyerEmail': basket.owner.email,
+            'buyerFullName': basket.owner.full_name,
+            'amount': str(basket.total_incl_tax),
+            'responseUrl': self.responseUrl,
+            'confirmationUrl': self.confirmationUrl,
+        }
+
+        single_seat = self.get_single_seat(basket)
+
+        if single_seat:
+            parameters['description'] = single_seat.course_id
+        else:
+            parameters['description'] = 'default_test_description'
+
+        parameters['referenceCode'] = basket.order_number
+        parameters['signature'] = self._generate_signature(parameters)
+
+        return parameters
+
+    @staticmethod
+    def get_single_seat(basket):
+        """
+        Return the first product encountered in the basket with the product
+        class of 'seat'.  Return None if no such products were found.
+        """
+        try:
+            seat_class = ProductClass.objects.get(slug='seat')
+        except ProductClass.DoesNotExist:
+            # this occurs in test configurations where the seat product class is not in use
+            return None
+
+        for line in basket.lines.all():
+            product = line.product
+            if product.get_product_class() == seat_class:
+                return product
+
+        return None
+
+    def handle_processor_response(self, response, basket=None):
+        """
+        Handle a response (i.e., "merchant notification") from PayU.
+
+        This method does the following:
+            1. Verify the validity of the response.
+            2. Create PaymentEvents and Sources for successful payments.
+
+        Arguments:
+            response (dict): Dictionary of parameters received from the payment processor.
+
+        Keyword Arguments:
+            basket (Basket): Basket being purchased via the payment processor.
+
+        Raises:
+            TransactionDeclined: Indicates the payment was declined by the processor.
+            GatewayError: Indicates a general error on the part of the processor.
+            InvalidCyberSourceDecision: Indicates an unknown decision value
+        """
+
+        # Validate the signature
+        if not self.is_signature_valid(response):
+            raise InvalidSignatureError
+
+        # Raise an exception for payments that were not accepted. Consuming code should be responsible for handling
+        # and logging the exception.
+        decision = response['state_pol']
+        if decision != '4':
+            exception = {
+                '6': TransactionDeclined,
+                '104': GatewayError
```

Comment:
here the same confusion arises. I think we can either leave a link to http://developers.payulatam.com/en/web_checkout/variables.html or copy some of the variables.
Another option would be to set some class or module constants (or enum). Perhaps only the constants are enough